tracer/src/Datadog.Trace/Debugger/Snapshots/Redaction.cs

```diff
@@ -306,9 +306,14 @@ private static bool IsRemovableChar(char c)
 
         public static void SetConfig(DebuggerSettings settings)
         {
-            foreach (var identifier in settings.RedactedIdentifiers)
+            if (settings.RedactedIdentifiers.Count > 0)
             {
-                _redactKeywords.Add(Normalize(identifier.Trim()));
+                _redactKeywords.UnionWith(settings.RedactedIdentifiers.Select(x => Normalize(x.Trim())));
```

Comment:
We try to avoid using LINQ (`Select()`) for performance reasons. This will allocate a delegate and a few iterators. Consider using `foreach` and `_redactKeywords.Add()`

Reply:
Also, I can't comment here, but in `Normalize()` there's a line:
```
sb.Append(name.Substring(startIndex: 0, i));
```
This `Substring()` allocates a new string. There is a `StringBuilder.Append()` overload that avoids this:
```
sb.Append(name, startIndex: 0, count: i);
```

Reply:
Agree. I "cheated" here because it only happens once and the code is very readable that way. But I agree with the point about startup time and serverless. I'll fix that.

Reply:
Fixed in 72e1e6102f92849dc141300a7685b15833ea4e7a

Reply:
Thanks!